packages/pretty-format/src/plugins/dom_collection.js

```diff
@@ -42,23 +39,36 @@ export const serialize = (
   refs: Refs,
   printer: Printer,
 ): string => {
+  const name = collection.constructor.name;
   if (++depth > config.maxDepth) {
-    return '[' + collection.constructor.name + ']';
+    return '[' + name + ']';
   }
 
   return (
-    collection.constructor.name +
-    SPACE +
-    '{' +
-    printObjectProperties(
-      convertCollectionToObject(collection),
-      config,
-      indentation,
-      depth,
-      refs,
-      printer,
-    ) +
-    '}'
+    (config.min ? '' : name + SPACE) +
+    (OBJECT_NAMES.indexOf(name) !== -1
+      ? '{' +
+        printObjectProperties(
+          name === 'NamedNodeMap'
+            ? Array.prototype.reduce.call(collection, propsReducer, {})
+            : Object.assign({}, collection),
+          config,
+          indentation,
+          depth,
+          refs,
+          printer,
+        ) +
+        '}'
+      : '[' +
+        printListItems(
+          Array.prototype.slice.call(collection),
```

Comment:
`Array.from`?

Reply:
Yes, I will leave the compiling to Babel :)